packages/headless/src/utils/validate-payload.ts

```diff
@@ -47,11 +48,16 @@ export const validatePayloadAndThrow = <P>(
     };
   }
 
-  return {
-    payload: new Schema(definition as SchemaDefinition<Required<P>>).validate(
-      payload
-    ) as P,
-  };
+  const asRecordValue = new RecordValue({
+    options: {required: true},
```

Comment:
That's the one thing I'm not 100% sure: This could in theory be breaking, as we are now validating that we are at least receiving "something" (an object) and that it must exists (required: true).

Before that, someone could dispatch an action without any parameter when it should "normally" receive one (eg: updateBasicConfiguration()). That would have been a no-op, but no error would be thrown.

With this change, it is now an error to do that.
What do you guys think ?

Reply:
I think it should be fine because even following these changes, the dispatch should still be a no-op.

When an error is thrown, it gets caught, serialized, and stored as an `error` property on the action object.
A middleware checks if an `error` property exists. If found, it logs the error, and returns early.

So if I were to summarize the behaviour:

Before: no-op
After: no-op, with error logged to the console.